web/unit_tests/test_utils.js

```diff
@@ -30,35 +46,73 @@ var setupKMW = function(ui) {
    */
   window.setTimeout(function() {
     initFunc();
-  }, 5);
+  }, 8);
 }
 
-var setupScript = function(src) {
+var setupScript = function(src, done, timeout, uiInitCheck) {
   var Lscript = document.createElement('script');
   Lscript.charset="UTF-8";        // KMEW-89
   Lscript.type = 'text/javascript';
   Lscript.async = false;
+  if(done) {
+    Lscript.onload = initTimer(done, timeout, uiInitCheck);
+  }
 
   Lscript.src = src;
 
   return Lscript;
 }
 
 var teardownKMW = function() {
+  keyman['shutdown']();
   delete window.keyman;
   window["keyman"] = undefined;
 }
 
 // Make sure the main script loads...
-var initTimer = function(done, timeout) {
-  window.setTimeout(function() {
-    if(window['keyman']) {
-      // ... and then give KMW and the UI a bit more time to init and attach.
+var initTimer = function(done, timeout, uiInitCheck) {
+  var uiLoadDelay;
+  if(typeof(uiInitCheck) != 'function') {
+    uiInitCheck = function() { return true; };
+    uiLoadDelay = false;
+  } else {
+    uiLoadDelay = true;
+  }
+
+  // We need managed state for this.
+  var InitializationManager = function() {
+    this.killSwitch = false;
+
+    this.initCheckCallback = function() {
+      if(window['keyman'] && window['keyman'].initialized == 2 && uiInitCheck()) {
+        if(done) {
+          this.timer = window.setTimeout(function() { 
+            // There can be some odd cross-interference with the UI modules and their initialization.
```

Comment:
It'd be good to quantify this. It's all in our grasp; we just need to map it out :).

Reply:
It would be nice, yes.  I spent a couple of hours yesterday trying to fine-tune things and find the exact source of the issues, but never quite figured out how the cross-interference was happening.  It may be more trouble than it's worth at the moment; I suppose a new issue for it is in order?

Reply:
The most fun instance was where the Float UI was initializing, the window['keyman'] instance of KeymanWeb said the Float UI was initiailizing, but the instance of window['keyman'] the Float UI saw said the Button UI was correct.  So, Float's `initialize()` was being called without its supporting functions... somehow.